sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -326,4 +327,17 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case _ => Nil
     }
   }
+
+  case class DDLStrategy(context: SQLContext) extends Strategy {
+    def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
+      case describe: logical.DescribeCommand =>
+        val resolvedTable = context.executePlan(describe.table).analyzed
+        resolvedTable match {
+          case LogicalRelation(relation) =>
+            Seq(ExecutedCommand(DDLDescribeCommand(relation, describe.output, describe.isExtended)))
+          case o: logical.DescribeCommand =>
+            ExecutedCommand(DescribeCommand(planLater(o), describe.output)) :: Nil
```

Comment:
Maybe we do not need this strategy. In ddl.scala we parse a runnable ```DDLDescribeCommand``` not logical.DescribeCommand, then it will follow commandstrategy
```
case object CommandStrategy extends Strategy {
    def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
      case r: RunnableCommand => ExecutedCommand(r) :: Nil
```